src/python/WMCore/BossAir/Plugins/PyCondorPlugin.py

```diff
@@ -972,16 +972,29 @@ def customizePerJob(self, job):
 
         # Performance estimates
         if job.get('estimatedJobTime'):
-            jdl.append('+MaxWallTimeMins = %d\n' % int(job['estimatedJobTime'] / 60.0))
+            estimatedMins = int(job['estimatedJobTime'] / 60.0)
+            origCores = int(job.get('numberOfCores', 1))
+            estimatedMinsSingleCore = estimatedMins*origCores
+            jdl.append('+MaxWallTimeMins = WMCore_ResizeJob ? (%d/RequestCpus + 60) : %d\n' % (estimatedMinsSingleCore, estimatedMins))
+            
         if job.get('estimatedMemoryUsage'):
-            jdl.append('request_memory = %d\n' % int(job['estimatedMemoryUsage']))
+            extraMem = int(job.get('extraMemoryPerCore', 500))
```

Comment:
500MB extra per core sounds about right and is based on feedback from the CMSSW team. Tier0 uses 2500 + (n-1) * 500MB and we never had problems in (proton proton) PromptReco. HeavyIon is a different matter though and I would expect some upgrade workflows might need more too.